Xplat/src/main/java/vazkii/botania/common/item/GrassSeedsItem.java

```diff
@@ -59,44 +70,54 @@ public GrassSeedsItem(IslandType type, Properties props) {
 	public InteractionResult useOn(UseOnContext ctx) {
 		Level world = ctx.getLevel();
 		BlockPos pos = ctx.getClickedPos();
-		BlockState state = world.getBlockState(pos);
 		ItemStack stack = ctx.getItemInHand();
 
+		return applySeeds(world, pos, stack)
+				? InteractionResult.sidedSuccess(world.isClientSide())
+				: InteractionResult.PASS;
+	}
+
+	public boolean applySeeds(Level world, BlockPos pos, ItemStack stack) {
```

Comment:
An antipattern I've been trying to move away from is interaction methods that return bools. Instead, keep the existing InteractionResult return values. In `useOn`, just return it through, and in the `DispenserBehavior`, use `shouldConsume` to convert it to a bool near the return site. That way the result value stays as a meaningful enum as long as possible before turning into a meaningless bool.